src/core/lib/gprpp/no_destruct.h

```diff
@@ -77,6 +80,9 @@ class NoDestructSingleton {
   static T* Get() { return &*value_; }
```

Comment:
I'd suggest clarifying in the comments above that there will be exactly one instance *of this type* globally for the lifetime of the process. The implementation is straightforward, makes it clear, but the bit about "when a program desires a single instance of a type" could mean a single instance in some other scope.

I'd also recommend a test to show basic usage.